### PR TITLE
Stabilize TestWorkflowTermination

### LINKED ISSUE #1095: TestWorkflowTermination is unstable
Parts of TestWorkflowTermination is based on Thread sleep and is unstable

### PR DESCRIPTION
**Pull request by alirezazamani**
### Issues
- [x] My PR addresses the following Helix issues and references them in the PR title:
Fixes #1095 

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
In this PR, the Thread sleeps have been removed from TestWorkflowTermination and replaced with TestHelper verify method.
Also, testWorkflowJobFail is polling for the Failed state for 10 seconds. However, the tasks in job3 of this workflow need to be retried for 10 times and then this job will go to Failed state. However, 10 seconds might be too short for this case. Hence, in this PR the maxNumberOfAttempts has been set to 5 for the tasks.

### Tests
- [x] The following is the result of the "mvn test" command on the appropriate module:

helix-core:
```
[ERROR] Failures: 
[ERROR]   TestJobQueueCleanUp.testJobQueueAutoCleanUp » ThreadTimeout Method org.testng....
[INFO] 
[ERROR] Tests run: 1139, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:26 h
[INFO] Finished at: 2020-06-15T16:12:13-07:00
[INFO] ------------------------------------------------------------------------
```
The failed test succeeded when run individually.
mvn test -Dtest="TestJobQueueCleanUp"
```
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 14.574 s - in org.apache.helix.integration.task.TestJobQueueCleanUp
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  20.532 s
[INFO] Finished at: 2020-06-15T16:43:01-07:00
[INFO] ------------------------------------------------------------------------
```
helix-rest:
```
[INFO] Tests run: 163, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 47.106 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 163, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  52.804 s
[INFO] Finished at: 2020-06-15T16:44:19-07:00
[INFO] ------------------------------------------------------------------------
```

### Commits

- [x] My commits all reference appropriate Apache Helix GitHub issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- [x] My diff has been formatted using helix-style.xml

### CODE REVIEW

File: helix-core/src/test/java/org/apache/helix/integration/task/TestWorkflowTermination.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ public void beforeClass() throws Exception {
     String taskState = shouldJobFail ? TaskState.FAILED.name() : TaskState.COMPLETED.name();
     return new JobConfig.Builder().setTargetResource(WorkflowGenerator.DEFAULT_TGT_DB)
         .setTargetPartitionStates(Sets.newHashSet(MasterSlaveSMD.States.MASTER.name()))
-        .setWorkflow(workflow).setCommand(MockTask.TASK_COMMAND)
+        .setWorkflow(workflow).setCommand(MockTask.TASK_COMMAND).setMaxAttemptsPerTask(5)
```
Comment (dasahcc) [inline helix-core/src/test/java/org/apache/helix/integration/task/TestWorkflowTermination.java:65]: Why we want to limit this?

**Comment by alirezazamani** [comment]
testWorkflowJobFail is polling for the Failed state for 10 seconds. However, I noticed the tasks in job3 of this workflow need to be retried for 10 times and then this job will go to Failed state. However, 10 seconds is little bit short for this case (for other jobs 1 and 2 and possibly 4 to run and job 3 to be retried 10 times and then failed). I noticed that the job fails right after 10 seconds. Hence, in this PR the maxNumberOfAttempts has reduced set to 5 for the tasks.

File: helix-core/src/test/java/org/apache/helix/integration/task/TestWorkflowTermination.java (hunk 1 above)
Comment (alirezazamani) [reply helix-core/src/test/java/org/apache/helix/integration/task/TestWorkflowTermination.java:65]: testWorkflowJobFail is polling for the Failed state for 10 seconds. However, I noticed the tasks in job3 of this workflow need to be retried for 10 times and then this job will go to Failed state. However, 10 seconds is little bit short for this case (for other jobs 1 and 2 and possibly 4 to run and job 3 to be retried 10 times and then failed). I noticed that the job fails right after 10 seconds. Hence, in this PR the maxNumberOfAttempts has reduced set to 5 for the tasks.

**Comment by alirezazamani** [comment]
This PR is ready to be merged, approved by @dasahcc 

Final commit message:
Stabilize TestWorkflowTermination

In this PR, TestWorkflowTermination has been stabilized.
Also, testGetStateModelDef which has not been adjusted
in previous commits is fixed.

**Pull request merged**